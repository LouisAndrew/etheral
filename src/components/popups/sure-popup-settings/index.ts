
import { SurePopupSettings } from './sure-popup-settings';

export default SurePopupSettings;