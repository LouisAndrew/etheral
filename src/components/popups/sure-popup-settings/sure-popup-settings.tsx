import React from 'react';

import { Box, Button, Heading, Text, Flex } from 'rebass';
import { Icon } from '@iconify/react';
import closeFill from '@iconify/icons-ri/close-fill';

type Props = {
    yes: () => void;
    no: () => void;
    close: () => void;
};

const SurePopupSettings: React.FC<Props> = ({ yes, no, close }) => {
    return (
        <Box bg="white.0" p={[7]} sx={{ position: 'relative' }}>
            <Box
                onClick={close}
                sx={{
                    position: 'absolute',
                    top: 4,
                    right: 4,
                    '&:hover': { cursor: 'pointer' },
                    '& svg': { width: 24, height: 24 },
                }}
            >
                <Icon icon={closeFill} />
            </Box>
            <Heading variant="h2" mb={[3]}>
                Are you sure you want to update the website now?
            </Heading>
            <Text variant="body" color="black.1">
                By clicking &apos;yes&apos; the website would be automatically
                updated and ready to browse in about 5 minutes
            </Text>
            <Flex mt={[3]} justifyContent="space-between">
                <Button onClick={yes} width="49%">
                    Yes
                </Button>
                <Button onClick={no} bg="misc.discount" width="49%">
                    No
                </Button>
            </Flex>
        </Box>
    );
};

export { SurePopupSettings };
