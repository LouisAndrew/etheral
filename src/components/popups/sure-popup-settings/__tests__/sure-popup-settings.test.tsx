import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import userEvent from '@testing-library/user-event';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';

import { SurePopupSettings } from '../sure-popup-settings';

describe('SurePopupSettings', () => {
    const yes = jest.fn(() => {});
    const no = jest.fn(() => {});
    const close = jest.fn(() => {});

    const Element = <SurePopupSettings yes={yes} no={no} close={close} />;

    afterEach(cleanup);

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(Element, div);
    });

    it('Should render and react correctly when user clicks the yes button', () => {
        const { getByText } = render(Element);
        const YesButton = getByText('Yes');
        userEvent.click(YesButton);
        expect(yes).toBeCalled();
    });

    it('Should render and react correctly when user clicks the yes button', () => {
        const { getByText } = render(Element);
        const NoButton = getByText('No');
        userEvent.click(NoButton);
        expect(no).toBeCalled();
    });

    it('matches snapshot', () => {
        const run = true;

        if (run) {
            const tree = renderer.create(Element).toJSON();
            expect(tree).toMatchSnapshot();
        }
    });
});
