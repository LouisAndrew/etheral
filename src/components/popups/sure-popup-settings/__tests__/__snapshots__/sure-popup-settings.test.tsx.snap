// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SurePopupSettings matches snapshot 1`] = `
<div
  className="css-113sbh6"
>
  <div
    className="css-lshl07"
    onClick={[MockFunction]}
  >
    <svg
      dangerouslySetInnerHTML={
        Object {
          "__html": "<path d=\\"M12 10.586l4.95-4.95l1.414 1.414l-4.95 4.95l4.95 4.95l-1.414 1.414l-4.95-4.95l-4.95 4.95l-1.414-1.414l4.95-4.95l-4.95-4.95L7.05 5.636z\\" fill=\\"currentColor\\"/>",
        }
      }
      focusable={false}
      height="1em"
      preserveAspectRatio="xMidYMid meet"
      style={
        Object {
          "transform": "rotate(360deg)",
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    />
  </div>
  <h2
    className="css-15jfn87"
  >
    Are you sure you want to update the website now?
  </h2>
  <div
    className="css-8898eo"
  >
    By clicking 'yes' the website would be automatically updated and ready to browse in about 5 minutes
  </div>
  <div
    className="css-18m6ozg"
  >
    <button
      className="css-78ookd"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [
              Class {
                "_dispatchInstances": null,
                "_dispatchListeners": null,
                "_targetInst": null,
                "altKey": null,
                "bubbles": null,
                "button": null,
                "buttons": null,
                "cancelable": null,
                "clientX": null,
                "clientY": null,
                "ctrlKey": null,
                "currentTarget": [Function],
                "defaultPrevented": null,
                "detail": null,
                "dispatchConfig": null,
                "eventPhase": null,
                "getModifierState": [Function],
                "isDefaultPrevented": [Function],
                "isPropagationStopped": [Function],
                "isTrusted": null,
                "metaKey": null,
                "movementX": [Function],
                "movementY": [Function],
                "nativeEvent": null,
                "pageX": null,
                "pageY": null,
                "relatedTarget": [Function],
                "screenX": null,
                "screenY": null,
                "shiftKey": null,
                "target": null,
                "timeStamp": [Function],
                "type": null,
                "view": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M10 15.172l9.192-9.193l1.415 1.414L10 18l-6.364-6.364l1.414-1.414z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      Yes
    </button>
    <button
      className="css-1m6x1fp"
      onClick={
        [MockFunction] {
          "calls": Array [
            Array [
              Class {
                "_dispatchInstances": null,
                "_dispatchListeners": null,
                "_targetInst": null,
                "altKey": null,
                "bubbles": null,
                "button": null,
                "buttons": null,
                "cancelable": null,
                "clientX": null,
                "clientY": null,
                "ctrlKey": null,
                "currentTarget": [Function],
                "defaultPrevented": null,
                "detail": null,
                "dispatchConfig": null,
                "eventPhase": null,
                "getModifierState": [Function],
                "isDefaultPrevented": [Function],
                "isPropagationStopped": [Function],
                "isTrusted": null,
                "metaKey": null,
                "movementX": [Function],
                "movementY": [Function],
                "nativeEvent": null,
                "pageX": null,
                "pageY": null,
                "relatedTarget": [Function],
                "screenX": null,
                "screenY": null,
                "shiftKey": null,
                "target": null,
                "timeStamp": [Function],
                "type": null,
                "view": null,
              },
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": undefined,
            },
          ],
        }
      }
    >
      <svg
        dangerouslySetInnerHTML={
          Object {
            "__html": "<path d=\\"M12 10.586l4.95-4.95l1.414 1.414l-4.95 4.95l4.95 4.95l-1.414 1.414l-4.95-4.95l-4.95 4.95l-1.414-1.414l4.95-4.95l-4.95-4.95L7.05 5.636z\\" fill=\\"currentColor\\"/>",
          }
        }
        focusable={false}
        height="1em"
        preserveAspectRatio="xMidYMid meet"
        style={
          Object {
            "transform": "rotate(360deg)",
            "verticalAlign": "-0.125em",
          }
        }
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      />
      No
    </button>
  </div>
</div>
`;
